app/views/pieces/update.js.erb

```diff
@@ -1,2 +1,4 @@
-$('.game-board').html('<%= j (render partial: 'games/board', the_game: @game) %>');
+$('.game-board').html('<%= j (render partial: 'games/board') %>');
+$('.player-one').html('<%= j (render partial: 'games/player_one') %>');
```

Comment:
For clarity, you are wanting to do a full-page reload for castling but trigger the piece#update action for en_passant?

Reply:
I dont't know if I would call it a full page reload. But for the buttons for Castling to show up more then the board has to be reloaded after every move. The buttons could be moved to the board but then still have the problem of captured pieces not showing up. This seemed like the simplest (though not most optimal) solution.

Reply:
At this point it is a full-page reload for the castling method. It does not call update.js.erb. Which is fine, like you said we can change it later. I just wasn't sure if you had intended to only reload the player1/player2 columns for en passant only or not. The board and the columns are being reloaded after every move at this point already.